factor/telegram/Telegram.php

```diff
@@ -0,0 +1,10 @@
+<?php
+$path = "https://api.telegram.org/bot1204730014:AAEMpzxxRPRnQTLZ9eCMNgCAmBd0d9pX8iQ";
```

Comment:
It would be good to keep this file but turn it into an admin only 'test connection' in the admin settings page where you set the api

Reply:
ah I'm not quite sure the purpose of this, but it looks like it is the real sending / api side fo things?

If so please move this into /classes/local/telegram.php and please also the the `new curl()` moodle library for all api calls so it can go through a proxy if needed

Reply:
hmm ok now I'm more confused as that already looks to be done but it wasn't working